### PR TITLE
Remove cruft

### LINKED ISSUE #33: invalid syntax on async keyword
When I flake8 the file from the command line everything passes, but I get invalid syntax from the plugin.

I'm using pyenv, but the shims seem to be loading correctly below, since it pulls the path from the global python version.

```
$ pyenv global
3.5.0
```

Output from editor open to file lint (conftest.py)

```
DPI scale: 1
startup, version: 3083 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/numberoverzero/.config/sublime-text-3/Packages
state path: /home/numberoverzero/.config/sublime-text-3/Local
zip path: /opt/sublime_text/Packages
zip path: /home/numberoverzero/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.10723
startup time: 0.205469
first paint time: 0.255404


<SNIP> reloading plugin spam </SNIP>


reloading plugin SublimeLinter-flake8.linter
SublimeLinter: flake8 linter loaded 
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: temp directory: /tmp/SublimeLinter3-numberoverzero 
SublimeLinter: find_python(version='3', script=None, module=None) 
SublimeLinter: find_python: requested version = {'minor': None, 'major': 3} 
SublimeLinter: user shell: /bin/bash 
SublimeLinter: computed PATH using /bin/bash:
/home/numberoverzero/.pyenv/shims
/home/numberoverzero/.pyenv/bin
/usr/local/sbin
/usr/local/bin
/usr/sbin
/usr/bin
/sbin
/bin
/usr/games
/usr/local/games
/usr/local/bin
/home/numberoverzero/bin
~/npm/bin
/home/numberoverzero/.rvm/bin
 
SublimeLinter: find_posix_python: python3 => /home/numberoverzero/.pyenv/shims/python3 
SublimeLinter: find_python: system python = /home/numberoverzero/.pyenv/shims/python3 
SublimeLinter: find_python: available version = {'minor': 5, 'major': 3} 
SublimeLinter: find_python: <= ('/home/numberoverzero/.pyenv/shims/python3', None, 3, 5) 
SublimeLinter: sys.path for /home/numberoverzero/.pyenv/shims/python3:
/home/numberoverzero/ws/bloop
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/arrow-0.7.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/jmespath-0.9.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/python_dateutil-2.4.2-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/six-1.10.0-py3.5.egg
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python35.zip
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/plat-linux
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/lib-dynload
/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages
 
SublimeLinter: flake8 imported <module 'flake8.engine' from '/home/numberoverzero/.pyenv/versions/3.5.0/lib/python3.5/site-packages/flake8/engine.py'> 
Package Control: Skipping automatic upgrade, last run at 2016-01-12 19:34:04, next run at 2016-01-12 20:34:04 or after
SublimeLinter: flake8 version: 2.5.1 
SublimeLinter: flake8: (>= 2.2.2) satisfied by 2.5.1 
SublimeLinter: flake8 activated: <builtin> 
SublimeLinter: chdir not set or invalid, using /home/numberoverzero/ws/bottom/tests 
SublimeLinter: flake8: conftest.py <builtin> 
SublimeLinter: flake8 options: {'reporter': <class 'SublimeLinter-flake8.linter.Flake8.get_report.<locals>.Report'>, 'jobs': '1', 'max_complexity': -1} 
SublimeLinter: flake8 output:
conftest.py:22:14: E901 SyntaxError: invalid syntax 

```

### LINKED ISSUE #34: SublimeLinter is calling Flake8 with the system python instead of the Virtualenv
I install flake8 in a virtualenv using [pipsi](https://github.com/mitsuhiko/pipsi).  This sets up a bin directory in my path to call python scripts but keeping them isolated in virtualenvs.

It works fine when flake8 is run normally:
```
kstone@kstone-mbp ~> which flake8
$HOME/.local/bin/flake8
kstone@kstone-mbp ~> flake8 --version
2.5.0 (pep8: 1.5.7, pyflakes: 1.0.0, mccabe: 0.3.1) CPython 2.7.10 on Darwin
```

But SublimeLinter uses the system python (if another one isn't configured) rather than calling the entry point directly and thus doesn't register the appropriate virtualenv:
```
SublimeLinter: flake8 version query: /usr/bin/python $HOME/.local/bin/flake8 --version 
SublimeLinter: WARNING: no flake8 version could be extracted from:
Traceback (most recent call last):
  File "$HOME/.local/bin/flake8", line 7, in <module>
    from flake8.main import main
ImportError: No module named flake8.main
```

Basically it's calling `python ~/.local/bin/flake8` instead of just executing `~/.local/bin/flake8`

### LINKED ISSUE #35: show-code option is not working
Setting `show-code` option doesn't work, breaking also the whole linting.
Here is the error from console:

SublimeLinter: flake8: vminfo_quick.py ['/usr/bin/python2.7', '/usr/local/bin/flake8', '--ignore=F403,E111,E113', '--max-line-length=100', '--jobs=1', '--max-complexity=-1', '--show-code=True', '-'] 
SublimeLinter: flake8 output:
Usage: flake8 [options] input ...

flake8: error: no such option: --show-code

### LINKED ISSUE #36: Adding custom flake8 args in Python3
Hey!

I've just bumped up to Python3 and have found that I can no longer pass extra args to flake8 via the SublimeLinter plugin.

I'm currently defining args in my SublimeLinter prefs as so:

```
"args": [
    "--radon-max-cc",
    "6"
]
```

Looking at the logs in debug mode - Python2 shows something like this:

`SublimeLinter: flake8: list_.py ['/python2.7/bin/python', '/python2.7/bin/flake8', '--radon-max-cc', '1', '--max-complexity=-1', '--jobs=1', '--max-line-length=100', '--ignore=P101,P102,P103', '-']`

However, in Python3, it looks to be operating in a totally different manner:

```
SublimeLinter: flake8: list_.py <builtin> 
SublimeLinter: flake8 options: {'max_line_length': 100, 'jobs': '1', 'reporter': <class 'SublimeLinter-flake8-master.linter.Flake8.get_report.<locals>.Report'>, 'max_complexity': -1, 'ignore': ['P101', 'P102', 'P103']}

```
I'm struggling to see how I should be passing through the custom arguments!

Any help would be much appreciated.

Cheers,
Ian

### PR DESCRIPTION
**Pull request by kaste**
* Never use builtin python 3.3 for linting Fixes #33
* Make `show-code` actually work Fixes #35
* Add option `executable` so you can point explicitly to a flake8 installation in case SL isn't smart enough (which it isn't on Windows)

Fixes #34, fixes #36

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -56,83 +54,11 @@ class Flake8(PythonLinter):
         '--max-line-length=': None,
         '--max-complexity=': -1,
         '--jobs=': '1',
-        '--show-code=': False
+        'show-code': False,
```
Comment (kaste) [inline linter.py]: That's essentially the thing to fix #35. Double '--' means that this is an argument for flake8, but we want it to be an internal argument for the linter.

File: linter.py [hunk 2]
```diff
@@ -56,83 +54,11 @@ class Flake8(PythonLinter):
...
+        'executable': ''
     }
-    inline_settings = ('max-line-length', 'max-complexity', 'show_code')
+    inline_settings = ('max-line-length', 'max-complexity')
     inline_overrides = ('select', 'ignore', 'builtins')
-    module = 'flake8.engine'
-    check_version = True
```
Comment (kaste) [inline linter.py]: We don't want to use the python 3.3 that comes with sublime.

File: linter.py [hunk 3]
```diff
@@ -56,83 +54,11 @@ class Flake8(PythonLinter):
...
-    report = None
-    show_code = False
-    pyflakes_checker_module = None
-    pyflakes_checker_class = None
-
-    @classmethod
-    def initialize(cls):
-        """Initialize the class after plugin load."""
-
-        super().initialize()
```
Comment (kaste) [inline linter.py]: This call already fails!

File: linter.py [hunk 4]
```diff
@@ -56,83 +54,11 @@ class Flake8(PythonLinter):
...
-                check = entry.load()
-
-                if check.name == 'pyflakes':
-                    from pyflakes import checker
-                    cls.pyflakes_checker_module = checker
-                    cls.pyflakes_checker_class = check
-                    break
-
-    def check(self, code, filename):
-        """Run flake8 on code and return the output."""
```
Comment (kaste) [inline linter.py]: Not needed, we use an external flake8 executable

File: linter.py [hunk 5]
```diff
@@ -148,35 +74,17 @@ def split_match(self, match):
         if near:
             col = None
 
-        if self.show_code:
+        if self.get_view_settings(self).get('show-code'):
             message = ' '.join([error or warning or '', message])
```
Comment (kaste) [inline linter.py]: `get_view_settings` should just be enough.

File: linter.py [hunk 6]
```diff
@@ -148,35 +74,17 @@ def split_match(self, match):
...
         return match, line, col, error, warning, message, near
 
-    def get_report(self):
-        """Return the Report class for use by flake8."""
```
Comment (kaste) [inline linter.py]: unused

File: linter.py [hunk 7]
```diff
@@ -148,35 +74,17 @@ def split_match(self, match):
...
-                    return results
-
-            self.__class__.report = Report
-
-        return self.report
+    def build_cmd(self, cmd=None):
+        """Return a tuple with the command line to execute."""
+        
+        executable = self.get_view_settings().get('executable', None)
+        if executable:
```
Comment (kaste) [inline linter.py]: If the user provides an executable via settings just use it.

File: linter.py (hunk 5 above)
Comment (kaste) [inline linter.py]: (Shit: What's doing the self here?) :lollipop:

**Comment by kaste** [comment]
Well, can you make the linter happy?

**Comment by kaste** [comment]
image

:fireworks:

**Comment by Gr1N** [comment]
Really nice patch! So fix build and merge to master :)

**Comment by kaste** [comment]
Ok, what's going on here with travis? @groteworld

**Comment by jmagnusson** [comment]
Yippie! Can't wait for a new release with this! 🎊 🍻 🎉

File: linter.py [hunk 8]
```diff
@@ -10,12 +10,11 @@
...
 """This module exports the Flake8 plugin linter class."""
 
-import os
-from SublimeLinter.lint import persist, PythonLinter
+from SublimeLinter.lint import PythonLinter
 
 
 class Flake8(PythonLinter):
-
+
```
Comment (numberoverzero) [inline linter.py]: This is the only line causing the travis failure:

```
The command "flake8 . --max-line-length=120" exited with 1.
$ pep257 . --ignore=D202
./linter.py:16 in public class `Flake8`:
        D211: No blank lines allowed before class docstring (found 1)
```

Can you remove this so the build passes?  I'd love to see this merged :+1:

Comment (kaste) [inline linter.py]: I actually either get a  `D211: No blank lines allowed before class docstring (found 1)` https://travis-ci.org/SublimeLinter/SublimeLinter-flake8/builds/129267208  or a `D203: 1 blank line required before class docstring (found 0)` https://travis-ci.org/SublimeLinter/SublimeLinter-flake8/builds/129247772

**Comment by kaste** [comment]
Ah, referring to https://github.com/PyCQA/pydocstyle/issues/141 we loose the default ignore D203 as soon as we pass our own ignore info to pep257.

File: linter.py [hunk 9]
```diff
@@ -148,35 +74,17 @@ def split_match(self, match):
         if near:
             col = None
 
-        if self.show_code:
+        if self.get_view_settings().get('show-code'):
```
Comment (kaste) [inline linter.py]: `get_view_settings` should be enough

**Comment by kaste** [comment]
Okay, we're green. :raised_hands:

**Comment by jmagnusson** [comment]
😀

**Comment by FichteFoll** [comment]
Again, referring to https://github.com/PyCQA/pydocstyle/issues/141, you should be using the `add-ignore` option, which SublimeLinter-pydocstyle (name change) now supports: https://github.com/SublimeLinter/SublimeLinter-pydocstyle/commit/b17b4cdf8eb254cd403447a31033d5d41fa26a7b, https://github.com/SublimeLinter/SublimeLinter-pydocstyle/issues/11

File: linter.py [hunk 10]
```diff
@@ -56,83 +54,11 @@ class Flake8(PythonLinter):
         '--max-line-length=': None,
         '--max-complexity=': -1,
         '--jobs=': '1',
-        '--show-code=': False
+        'show-code': False,
+        'executable': ''
```
Comment (FichteFoll) [inline linter.py]: Shouldn't you also define a `executable` class attribute when you're removing `module`, or is this solely to circumvent https://github.com/SublimeLinter/SublimeLinter3/issues/455?

I would much rather not have to specify the flake8 executable path in my settings if SL is technically capable of locating it by itself.

**Comment by kaste** [comment]
Rebased!

**Comment by kaste** [comment]
Does anybody know the/one of the maintainer(s)? B/c from my experience it is either you get a response relatively quickly or never. And after two weeks we're heading towards never.

**Comment by Gr1N** [comment]
@groteworld can help you with this pr?

**Comment by FichteFoll** [comment]
push

This plugin is unusable for 3.5 code without this PR.

**Comment by EmilStenstrom** [comment]
Great fix! 🎊 Will my local version of SublimeLinter-flake8 be updated automatically or does there have to be a release?

**Comment by FichteFoll** [comment]
There already is a release. Package control will update it automatically on the next st restart.

**Comment by Gr1N** [comment]
Guys, something goes wrong after update.
Sublime console:
```
SublimeLinter: WARNING: no flake8 version could be extracted from:
Traceback (most recent call last):
  File "/usr/local/bin/flake8", line 7, in <module>
    from flake8.main import main
ImportError: No module named flake8.main```

And my flake8 version and path:
```
~
➔ flake8 --version
2.5.4 (pep8: 1.7.0, pyflakes: 1.2.3, mccabe: 0.4.0) CPython 3.5.1 on Darwin

~
➔ which flake8
/usr/local/bin/flake8
```
Any ideas?

**Comment by EmilStenstrom** [comment]
I get the same error as @Gr1N, I think the offending row is the debug line just above what was posted:

```
SublimeLinter: flake8 version query: /usr/local/bin/python /virtualenvs/projectX/bin/flake8 --version 
SublimeLinter: WARNING: no flake8 version could be extracted from:
Traceback (most recent call last):
  File "/virtualenvs/projectX/bin/flake8", line 7, in <module>
    from flake8.main import main
ImportError: No module named flake8.main
```

Note the prefix "/usr/local/bin/python". Doing the same call in the terminal reproduces it:
```
~ $ /usr/local/bin/python /virtualenvs/projectX/bin/flake8 --version 
Traceback (most recent call last):
  File "/virtualenvs/projectX/bin/flake8", line 7, in <module>
    from flake8.main import main
ImportError: No module named flake8.main
```

Removing the python executable works fine:
```
~ $ /virtualenvs/projectX/bin/flake8 --version 
2.5.4 (pep8: 1.7.0, pyflakes: 1.0.0, mccabe: 0.4.0) CPython 3.4.1 on Darwin
```

**Comment by kaste** [comment]
Damn it. Works for me.

**Comment by kaste** [comment]
Can you post your configuration. On my system I have python 2.7 and 3.5, mostly everything in virtualenvs. And it just works. Default python is 2.7, and a globally installed flake8.

**Comment by Gr1N** [comment]
I have python 2.7 (default) and 3.5 and I use virtualenvs too, but `flake8` installed globally using `pip3`. `/usr/local/bin/python` — python 2.7 in my system without `flake8`. Maybe python executable should be configurable through settings?

**Comment by EmilStenstrom** [comment]
I have python 2.7 as default (installed via homebrew) and Python 3.4 in paralell. I have flake installed inside a virtualenv running Python 3.4.

**Comment by kaste** [comment]
You sure have some linter config as well. ;-)

**Comment by kaste** [comment]
Could you try

```
diff --git a/linter.py b/linter.py
index 29bdb0c..d21b5e7 100644
--- a/linter.py
+++ b/linter.py
@@ -19,7 +19,7 @@ class Flake8(PythonLinter):
     """Provides an interface to the flake8 python module/script."""

     syntax = 'python'
-    cmd = ('flake8@python', '*', '-')
+    cmd = ('flake8', '*', '-')
     version_args = '--version'
     version_re = r'^(?P<version>\d+\.\d+\.\d+)'
     version_requirement = '>= 2.2.2'

```

**Comment by EmilStenstrom** [comment]
I'll try, but I've never modified a plugin file before. I'm on Mac OSX. Installed plugin through Package Control. Where can I find the linter.py file?

**Comment by FichteFoll** [comment]
Use https://packagecontrol.io/packages/PackageResourceViewer to access and modify a resource file.

**Comment by Gr1N** [comment]
@kaste your fix works perfect.
```SublimeLinter: flake8 version query: /usr/local/bin/flake8 --version 
SublimeLinter: flake8 version: 2.5.4 
SublimeLinter: flake8: (>= 2.2.2) satisfied by 2.5.4 
SublimeLinter: flake8 activated: /usr/local/bin/flake8 ```

**Comment by EmilStenstrom** [comment]
Thanks @FichteFoll, that was easier than I thought.

Modifying the file works great for me too. I guess my case is slightly different since I start sublime from inside a virtualenv where flake8 is installed. Non the less, it works! Thanks.

**Comment by kaste** [comment]
Some more feedback on my hotfix? I remember I had this in the first version of this PR, but I don't know anymore why I decided against it.

**Comment by kaste** [comment]
Follow up #41.

**Pull request merged**